### PR TITLE
Thk queues

### LINKED ISSUE #8: Add message queue mechanism
Add a message queue similar to 
https://github.com/gabime/spdlog#backtrace-support

This would temporarily store a fixed amount of non printed messages in a FIFO queue. A method should be created to "dump" all queued messages to the flexy realtime logs.

**The "dump" method could be triggered in multiple ways.**
 
1. A "DumpLogs" boolean tag on the ewon flexy. Setting this tag to 1 would trigger the log dump.
2. When "LOG_EXCEPTION()" is called the logs could dump. This would be useful for a scenerio where the log level is set quite low (ex: 1) but a serious issue happened and the user wants the full log context without having to replicate the issue with a different log level configured. 
3. A third party interface (Device Twin, Device Shadow, Custom REST interface).
4. etc.

### PR DESCRIPTION
**Pull request by TomKimsey**

### CODE REVIEW

File: src/com/hms_networks/americas/sc/logging/SocketLogger.java [hunk 1]
```diff
@@ -77,8 +78,17 @@ private static void OPEN() {
       try {
         socketServerConnection =
             (ServerSocketConnection) Connector.open("socket://:" + SOCKET_PORT);
-      } catch (Exception e) {
-        Logger.LOG_DEBUG("SocketLogger connection could not open.");
+      } catch (ConnectionNotFoundException e) {
+        Logger.LOG_DEBUG("SocketLogger connection could not open (ConnectionNotFoundException).");
```
Comment (alexjhawk) [inline src/com/hms_networks/americas/sc/logging/SocketLogger.java:82]: Is there a reason for not logging the exception as well?

File: README.md [hunk 2]
```diff
@@ -30,6 +31,15 @@ There are seven levels of configurable log level. The logging level is set by th
 | 1, -1 (Critical) | Critical application log messages (Startup, Initialization, Unrecoverable Error) |
 | 0 (None)         | Logging is disabled |
 
+## [Socket Logging](#table-of-contents)
+This library supports outputting log information over a raw socket connection on port 766.
+
+Netcat Example: Connect to a flexy with IP address 192.168.0.1
+```console
+nc 192.168.0.1 766
```
Comment (alexjhawk) [inline README.md]:
```suggestion
Netcat Example: Connect to a Flexy with IP address 10.0.0.53
```console
nc 10.0.0.53 766
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,15 @@
 # Ewon Flexy Logger Library Changelog
 
+## V1.2
```
Comment (alexjhawk) [inline CHANGELOG.md]:
```suggestion
## v1.2
```

to be consistent

**Review by alexjhawk — CHANGES_REQUESTED** [review]

File: src/com/hms_networks/americas/sc/logging/SocketLogger.java (hunk 1 above)
Comment (TomKimsey) [reply src/com/hms_networks/americas/sc/logging/SocketLogger.java:82]: I didnt want to eat up the queue space while its disconnected. The flexy seems to handle its sockets a bit strange so if you disconnect it often takes a few tries to restart the socketserver.

File: README.md (hunk 2 above)
Comment (TomKimsey) [reply README.md]: changed

File: CHANGELOG.md (hunk 3 above)
Comment (TomKimsey) [reply CHANGELOG.md]: changed

**Review by alexjhawk — APPROVED** [review]

**Review by oliver-walker-hms — APPROVED** [review]

**Pull request merged**